src/monitor.cpp

```diff
@@ -160,108 +160,211 @@ complex<double> fields_chunk::get_field(component c, const ivec &iloc) const {
+    
+    int my_stuff = E_stuff;
+    component comp_list[3];
+    if (is_electric(c)) {
+      comp_list[0] = Ex; comp_list[1] = Ey; comp_list[2] = Ez;
+      my_stuff = E_stuff;
+    }else if (is_magnetic(c)) {
+      comp_list[0] = Hx; comp_list[1] = Hy; comp_list[2] = Hz;
+      my_stuff = H_stuff;
+    } else if (is_D(c)) {
+      comp_list[0] = Dx; comp_list[1] = Dy; comp_list[2] = Dz;
+      my_stuff = D_stuff;
+    } else if (is_B(c)) {
+      comp_list[0] = Bx; comp_list[1] = By; comp_list[2] = Bz;
+      my_stuff = B_stuff;
+    }
+
+    std::complex<double> chi1_inv_tensor[9] = {std::complex<double>(1, 0),std::complex<double>(0, 0),std::complex<double>(0, 0),
+                                            std::complex<double>(0, 0),std::complex<double>(1, 0),std::complex<double>(0, 0),
+                                            std::complex<double>(0, 0),std::complex<double>(0, 0),std::complex<double>(1, 0)
+                                            };
+    std::complex<double> chi1_tensor[9] = {std::complex<double>(1, 0),std::complex<double>(0, 0),std::complex<double>(0, 0),
+                                        std::complex<double>(0, 0),std::complex<double>(1, 0),std::complex<double>(0, 0),
+                                        std::complex<double>(0, 0),std::complex<double>(0, 0),std::complex<double>(1, 0)
+                                            };
+
+    // Set up the chi1inv tensor with the DC components
+    for (int com_it=0; com_it<3;com_it++){
+      for (int dir_int=0;dir_int<3;dir_int++){
+        if (chi1inv[comp_list[com_it]][dir_int] )
+          chi1_inv_tensor[com_it + 3*dir_int] = chi1inv[comp_list[com_it]][dir_int][idx];
+      }
+    }
+    
+    matrix_invert(chi1_tensor, chi1_inv_tensor); // We have the inverse, so let's invert it.
+    
+    // ----------------------------------------------------------------- //
+    // ---- Step 2: Evaluate susceptibilities of each tensor element ---
+    // ----------------------------------------------------------------- //
+    
+    // loop over tensor elements
+    for (int com_it=0; com_it<3;com_it++){
+      for (int dir_int=0;dir_int<3;dir_int++){
+        std::complex<double> eps = chi1_tensor[com_it  + 3*dir_int];
+        component cc = comp_list[com_it];
+        direction dd = (direction)dir_int;
+        // Loop through and add up susceptibility contributions
+        // locate correct susceptibility list
+        susceptibility *my_sus = chiP[my_stuff];
+        while (my_sus) {
+          if (my_sus->sigma[cc][dd]) {
+            double sigma = my_sus->sigma[cc][dd][idx];
+            eps += my_sus->chi1(omega,sigma);
+          } 
+          my_sus = my_sus->next;
+        }
+
+        // Account for conductivity term
+        if (conductivity[cc][dd]) {
+          double conductivityCur = conductivity[cc][dd][idx];
+          eps = std::complex<double>(1.0, (conductivityCur/omega)) * eps;
```

Comment:
This line looks off to me — it seems like you are doing an element-wise product rather than a matrix multiplication?   

Since σ can only be a diagonal matrix currently, a matrix multiplication would multiply by `conductivity[cc][cc]` (the diagonal element from row `cc`), no?